### DISCUSSION TITLE
Is it possible to create a dynamic pages directory within a dynamic pages directory?

### DISCUSSION THREAD

**Discussion by es-ki** [remix-run/remix#6981] [Q&A] [answered]
😃Hello.
Can you please tell me if there is a way to set up an additional dynamic pages directory within the dynamic pages directory as shown below?
```
- app
  - routes
    - $categoryId
      - index.tsx
      - components
        - categoryList.tsx
      - $articleId
        - index.tsx
        - components
          - toc.tsx
```
In this directory configuration, accessing `$categoryId/$aricleId/index.tsx` results in a 404.
Accessing `$categoryId/index.tx`, the page appeared as expected.
Also, by creating the dynamic pages without separating them by directory as follows, the pages were displayed as expected in $articleId.tsx.
- $categoryId.tsx
- $categoryId_.$articleId.tsx

However, I would like to create dynamic pages by separating them in directories, because I want to separate the components and functions needed for each page.

**Comment by kiliman** [comment, marked as answer]
Remix `v2` routing only supports folders for colocation (that is, including your route module with non-route modules like components, images, styles, etc.)

If you need more flexibility, you can try `remix-flat-routes`. This package supports *hybrid* routes where you can use folders for organization as well as colocation.

https://github.com/kiliman/remix-flat-routes
